### PR TITLE
Removing more redundant objects in IBLevelSet/{ex1,ex2}. 

### PR DESCRIPTION
**Pull request by amneetb**
Now the examples give a feel of a truly Eulerian method

@drwells : There is a make clean error in IBlevelSet/ex2. Can you take a look?

amneetb@Amneets-MacBook-Pro:~/Softwares/IBAMR-Git/objs-dbg/examples/IBLevelSet/ex2$ make clean
rm -rf .libs _libs
rm -f main2d
if test "../../../../IBAMR" != "../../.." ; then \
	  rm -f ./input[2]d ; 
/bin/sh: -c: line 1: syntax error: unexpected end of file
make: *** [clean-local] Error 2

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [ ] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

File: examples/IBLevelSet/ex2/Makefile.am [hunk 1]
```diff
@@ -41,3 +41,5 @@ clean-local:
 	if test "$(top_srcdir)" != "$(top_builddir)" ; then \
 	  rm -f $(builddir)/input[2]d ;
```
Comment (drwells) [inline examples/IBLevelSet/ex2/Makefile.am:42]: @amneetb here's the problem - we need a trailing backslash on this line.

**Comment by drwells** [comment]
Nice. Would you mind, while we are here, fixing the ex2 problem?

Want to try updating `IBLevelSet/CMakeLists.txt` and writing `ex3/CMakeLists.txt`? We have a convenience macro for adding examples. The changes are essentially the same here as what I did in 8cb9de9963798d415ed074a0681fa2f5505ce953.

**Comment by drwells** [comment]
@amneetb If this is going to be in the release I need it to be merged this week - do you have time to get this ready?

**Comment by amneetb** [comment]
@drwells : Take a look.

**Comment by drwells** [comment]
Looks good. Can you handle the merge conflicts too?

**Comment by amneetb** [comment]
> Looks good. Can you handle the merge conflicts too?

Probably better if you or Aaron can handle it as it involves some recent changes related to machine precession checking. I'm not familiar with that change.

**Comment by amneetb** [comment]
uh precision not precession. You can tell that I'm a mechanical engineer.

**Comment by abarret** [comment]
> uh precision not precession. You can tell that I'm a mechanical engineer.

You should be using `IBTK::abs_equal_eps()` if you're comparing numbers that are close to zero; otherwise, you should use `IBTK::rel_equal_eps()`, which performs the same checks as `SAMRAI::equalEps()`.

**Comment by abarret** [comment]
I will rebase this and fix the conflicts.

**Comment by abarret** [comment]
@drwells and @amneetb: This has been rebased and merge conflicts fixed.

**Comment by drwells** [comment]
We need to duplicate the rest of the release process here too if we want to add this in late:
- check CMake and autotools with old and new dependencies
- update copyright
- clear any compiler warnings on maximally pedantic mode with GCC and clang
- check that examples compile with both autotools and CMake

**Comment by drwells** [comment]
@abarret are you willing to handle the release steps for this patch?

**Comment by abarret** [comment]
I'm not going to be able to get to this today or tomorrow. The earliest I can have it done is Friday.

**Comment by drwells** [comment]
I'll wrap it up and merge in that case today.

**Review by drwells — APPROVED** [review]

**Pull request merged**